craft_application/application.py

```diff
@@ -178,8 +178,7 @@ def _configure_services(
             work_dir=self._work_dir,
         )
 
-    @functools.cached_property
-    def project(self) -> models.Project:
+    def load_project(self, build_on: str, build_for: str) -> None:
```

Comment:
We have a moving target here, as while you were implementing this I changed `project` to this:
https://github.com/canonical/craft-application/blob/f584223edd3e56f78f59a7f4a0895a55ed1d117e/craft_application/application.py#L211

Reply:
This should be easy to reconcile once we settle on a grammar integration strategy.